neo/SmartContract/ApplicationEngine.cs

```diff
@@ -41,39 +76,31 @@ protected virtual long GetPrice()
                 case OpCode.SHA1:
```

Comment:
I think this should be removed as well, I think that it does not make sense to remove HASH160, HASH256 and leave the SHA here, @erikzhang.
@shargon @igormcoelho.

Let's reopen https://github.com/neo-project/neo-vm/pull/153

Reply:
I agree, because in the future SHA1 could be broken like md5, i think that we should move all these crypto logic to syscall, in order to prevent this kind of updates in vm